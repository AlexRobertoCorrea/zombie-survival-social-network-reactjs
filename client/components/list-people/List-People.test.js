import React from 'react';

import ListPeople from './List-people';

describe('<ListPeople />', () => {
  describe('Renders a circular progress because people was not loaded', () => {
    const store = {
      peopleStore: {
        fetchingData: true
      }
    };
    const classes = {};
    
    const listPeople = shallow(<ListPeople
      store={store}
      classes={classes}
    />);
    
    it('Ensure that listPeople works as expected', () => {
      expect(listPeople).toMatchSnapshot();
    });
  });
  
  describe('Renders the table of people correctly', () => {
    const store = {
      peopleStore: {
        people: []
      },
      fetchingData: false
    };
    const classes = {
      root: '',
      table: ''
    };
  
    const listPeople = shallow(<ListPeople
      store={store}
      classes={classes}
    />);
    
    it('Ensure that listPeople works as expected', () => {
      expect(listPeople).toMatchSnapshot();
    });
  });
});
