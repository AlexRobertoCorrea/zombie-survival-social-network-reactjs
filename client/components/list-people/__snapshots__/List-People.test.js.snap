// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPeople /> Renders a circular progress because people was not loaded Ensure that listPeople works as expected 1`] = `
<WithStyles(ListPeople)
  classes={Object {}}
  store={
    Object {
      "peopleStore": Object {
        "fetchingData": true,
      },
    }
  }
/>
`;

exports[`<ListPeople /> Renders the table of people correctly Ensure that listPeople works as expected 1`] = `
<WithStyles(ListPeople)
  classes={
    Object {
      "root": "",
      "table": "",
    }
  }
  store={
    Object {
      "fetchingData": false,
      "peopleStore": Object {
        "people": Array [],
      },
    }
  }
/>
`;
